### PR TITLE
fix(learn): update text as code for vector-dot-product.md

### LINKED ISSUE #40036: The term "nVectors" in "Rosetta Code: Vector dot product" is undefined
**Describe your problem and how to reproduce it:**
One of the test cases on the sidebar says `dotProduct(...nVectors) should return 156000`.  nVectors is undefined

Additionally, I notice that there are some tests where the text value doesn't match the actual test string.

**Add a Link to the page with the problem:**
https://www.freecodecamp.org/learn/coding-interview-prep/rosetta-code/vector-dot-product

**Tell us about your browser and operating system:**
* Browser Name: Firefox
* Browser Version: 82
* Operating System: Linux Mint

### PR DESCRIPTION
**Pull request by Karthik-Kashyap**
Checklist:

- [X] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [X] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [X] My pull request targets the `master` branch of freeCodeCamp.
- [X] All the files I changed are in the same world language, for example: only English changes, or only Chinese changes, etc.

Closes #40036 

1) Make changes to the text of the challenge.
2) Update one of the test case.

I have tested this in local.

### CODE REVIEW

File: curriculum/challenges/english/10-coding-interview-prep/rosetta-code/vector-dot-product.md
Comment (Sky020) [inline curriculum/challenges/english/10-coding-interview-prep/rosetta-code/vector-dot-product.md:27]:
```suggestion
  - text: <code>dotProduct()</code> should return <code>null</code>.
```

File: curriculum/challenges/english/10-coding-interview-prep/rosetta-code/vector-dot-product.md
Comment (Sky020) [inline curriculum/challenges/english/10-coding-interview-prep/rosetta-code/vector-dot-product.md:29]:
```suggestion
  - text: <code>dotProduct([[1], [1]])</code> should return <code>1</code>.
```

File: curriculum/challenges/english/10-coding-interview-prep/rosetta-code/vector-dot-product.md
Comment (Sky020) [inline curriculum/challenges/english/10-coding-interview-prep/rosetta-code/vector-dot-product.md:31]:
```suggestion
  - text: <code>dotProduct([[1], [1, 2]])</code> should return <code>null</code>.
```

File: curriculum/challenges/english/10-coding-interview-prep/rosetta-code/vector-dot-product.md
Comment (Sky020) [inline curriculum/challenges/english/10-coding-interview-prep/rosetta-code/vector-dot-product.md:33]:
```suggestion
  - text: <code>dotProduct([1, 3, -5], [4, -2, -1])</code> should return <code>3</code>.
```

File: curriculum/challenges/english/10-coding-interview-prep/rosetta-code/vector-dot-product.md
Comment (Sky020) [inline curriculum/challenges/english/10-coding-interview-prep/rosetta-code/vector-dot-product.md:35]:
```suggestion
  - text: <code>dotProduct([ 0, 1, 2, 3, 4 ], [ 0, 2, 4, 6, 8 ], [ 0, 3, 6, 9, 12 ], [ 0, 4, 8, 12, 16 ], [ 0, 5, 10, 15, 20 ])</code> should return <code>156000</code>.
```

File: curriculum/challenges/english/10-coding-interview-prep/rosetta-code/vector-dot-product.md
Comment (Sky020) [inline curriculum/challenges/english/10-coding-interview-prep/rosetta-code/vector-dot-product.md:25]:
```suggestion
  - text: <code>dotProduct</code> should be a function.
```

**Review by ShaunSHamilton — CHANGES_REQUESTED** [review]
Thank you, for getting this PR out.

While we are at it, could we change these to code tags as well. _Remember: when accepting suggestions on GitHub, they can be batched, in the 'Files changed' tab_

File: curriculum/challenges/english/10-coding-interview-prep/rosetta-code/vector-dot-product.md
Comment (Karthik-Kashyap) [reply curriculum/challenges/english/10-coding-interview-prep/rosetta-code/vector-dot-product.md]: Commited.

File: curriculum/challenges/english/10-coding-interview-prep/rosetta-code/vector-dot-product.md
Comment (Karthik-Kashyap) [reply curriculum/challenges/english/10-coding-interview-prep/rosetta-code/vector-dot-product.md]: Commited.

File: curriculum/challenges/english/10-coding-interview-prep/rosetta-code/vector-dot-product.md
Comment (Karthik-Kashyap) [reply curriculum/challenges/english/10-coding-interview-prep/rosetta-code/vector-dot-product.md]: Commited.

File: curriculum/challenges/english/10-coding-interview-prep/rosetta-code/vector-dot-product.md
Comment (Karthik-Kashyap) [reply curriculum/challenges/english/10-coding-interview-prep/rosetta-code/vector-dot-product.md]: Commited.

File: curriculum/challenges/english/10-coding-interview-prep/rosetta-code/vector-dot-product.md
Comment (Karthik-Kashyap) [reply curriculum/challenges/english/10-coding-interview-prep/rosetta-code/vector-dot-product.md]: Commited.

File: curriculum/challenges/english/10-coding-interview-prep/rosetta-code/vector-dot-product.md
Comment (Karthik-Kashyap) [reply curriculum/challenges/english/10-coding-interview-prep/rosetta-code/vector-dot-product.md]: Commited.

**Comment by Karthik-Kashyap** [comment]
This is my first time raising a PR for freeCodeCamp and in github. Why is it showing 'Changes requested, 1 review requesting changes by reviewers with write access.' ? Do I need to do anything?

**Comment by Sky020** [comment]
@Karthik-Kashyap > Do I need to do anything?

No worries. I am glad you asked. If a member of freeCodeCamp requests changes (as I did earlier), the PR has been reviewed. This means, the member who requested the changes (or another one) needs to approve the PR with its changes. If you click on the _"1 change requested"_ button/section, you will see who review-requested changes, and you are able to re-request a review.

For our general workflow, each PR requires 2 approving reviews, before it can be merged. For the most part, there is no need to re-request a review, even when GitHub presents the option, because maintainers of the repo are notified about all changes anyway.

If you have any more general questions, why not join us on either [Gitter](https://gitter.im/FreeCodeCamp/Contributors) or [Discord](https://discord.gg/pFspAhS), where we have dedicated Contributors channels.

**Comment by Sky020** [comment]
Before we approve this PR, I would like discussion on why it seems like the function should be written to be able to handle arrays of any depth...
```html
 <code>dotProduct([[1], [1]])</code> should return <code>1</code>.
```
and
```html
<code>dotProduct([1, 3, -5], [4, -2, -1])</code> should return <code>3</code>.
```

I am tempted to change the tests to only include arrays of `depth=1`. @moT01 Does this seem correct?

**Comment by RandellDawson** [comment]
@Sky020 It is totally valid to have multidemsional arrays for vectors.  They just must be the same size.

The following is total valid as they have the same size:
```
 <code>dotProduct([[1], [1]])</code> should return <code>1</code>
```

The following do not have the same size, which is why it should return `null`:
```
dotProduct([[1], [1, 2]])
```
I think this is what you were asking about?

I am fine with this PR being merged as it is now.

**Comment by Sky020** [comment]
> I think this is what you were asking about?

@RandellDawson Not really. I am asking whether this should be within the challenge - the requirement to handle multi-depth arrays, especially since the instructions do not mention it.

Should the camper's solution be able to handle?:
```javascript
[ [ [ [4, 5], [2, 3], [1, 2] ] ] ]
```

The given solution does not.

**Comment by moT01** [comment]
I was trying to understand this a little. There seems to be some discrepancy between the rosetta site and ours. The rosetta site says: `Create a function to compute the dot product of two vectors. If possible, make the vectors of arbitrary length.` Wikipedia says: `the dot product is an algebraic operation that takes two equal-length sequences of numbers (usually coordinate vectors), and returns a single number.` Wouldn't that mean that the function should only ever take two arrays of equal length. Anything else should return `null`.

The description on our site: `A vector is defined as having three dimensions as being represented by an ordered collection of three numbers: (X, Y, Z). Write a function that takes any numbers of vectors` - by that description, the function should only take arrays of length 3, but any number of them.

I may be misunderstanding some of this, but it seems like this could use a little rewrite to try and align it with what is on the rosetta site.

File: curriculum/challenges/english/10-coding-interview-prep/rosetta-code/vector-dot-product.md
Comment (moT01) [inline curriculum/challenges/english/10-coding-interview-prep/rosetta-code/vector-dot-product.md:29]:
```suggestion
  - text: <code>dotProduct([1], [1])</code> should return <code>1</code>.
```

File: curriculum/challenges/english/10-coding-interview-prep/rosetta-code/vector-dot-product.md
Comment (moT01) [inline curriculum/challenges/english/10-coding-interview-prep/rosetta-code/vector-dot-product.md:31]:
```suggestion
  - text: <code>dotProduct([1], [1, 2])</code> should return <code>null</code>.
```

**Review by moT01 — CHANGES_REQUESTED** [review]
I'm not sure if we want to keep the nested arrays, but my suggestions align the text with what is being tested. I think that should be fine.

**Review by moT01 — APPROVED** [review]
To try and get the remaining curriculum PR's merged, I have committed my suggestions. LGTM 🎉

**Comment by RandellDawson** [comment]
The original Rosetta Code challenge is located [here](https://rosettacode.org/wiki/Dot_product).  It is my understanding that vectors would always have three values (x, y, z).   Based on the problem scope (defined in our site and Rosetta Code's), I take that to mean, each vector is represented by a one-dimensional array like `[1, 0, -1]`.  [This PR](https://github.com/freeCodeCamp/freeCodeCamp/pull/40168) caught my attention in the sense that I believe `dotProduct([[1], [1]])` should actually return `null` instead of `1` since `[[1], [1]]` is a two-dimensional array containing two arrays instead of 3 elements (values).

**Review by ShaunSHamilton — DISMISSED** [review]
This LGTM 👍 

Congrats on your first pull request (PR)! 🎉

Thank you, @Karthik-Kashyap  for your contribution to the page! 👍
We are happy to accept these changes and look forward to future contributions. 📝

**Comment by RandellDawson** [comment]
See my updated comment above.

**Comment by ojeytonwilliams** [comment]
I'm not sure a dot product of 3 or more vectors really means anything, so I don't think we should be asking people to calculate them.  The last test doesn't really make sense to me, I think it should return null for that input.

**Comment by RandellDawson** [comment]
@ojeytonwilliams I guess that does make sense, since the dot product of two vectors is a scalar and you can not calculate the dot product of a scalar and a vector.

File: curriculum/challenges/english/10-coding-interview-prep/rosetta-code/vector-dot-product.md
Comment (Sky020) [inline curriculum/challenges/english/10-coding-interview-prep/rosetta-code/vector-dot-product.md:35]:
```suggestion
  - text: <code>dotProduct([ 0, 3, 6, 9, 12 ], [ 0, 4, 8, 12, 16 ])</code> should return <code>384</code>.
```

File: curriculum/challenges/english/10-coding-interview-prep/rosetta-code/vector-dot-product.md
Comment (Sky020) [inline curriculum/challenges/english/10-coding-interview-prep/rosetta-code/vector-dot-product.md:36]:
```suggestion
    testString: assert.equal(dotProduct([ 0, 3, 6, 9, 12 ], [ 0, 4, 8, 12, 16 ]), 384);
```

**Review by ShaunSHamilton — CHANGES_REQUESTED** [review]
Change this here, or just delete the last test completely?

**Comment by ojeytonwilliams** [comment]
> Write a function that takes two vectors (represented as one-dimensional arrays) as input and computes their dot product. Your function should return null on invalid inputs such as vectors of different lengths or passing anything other than two vectors.

Something like that.

**Comment by moT01** [comment]
Wondering if this is getting outside the scope of the issue. Certainly this could use some clarification - maybe another issue? I committed my suggestions in hopes to get it merged. But, if you want to fix it up - I guess I won't stop ya.

**Comment by RandellDawson** [comment]
@moT01 I think it is still within scope as we changed one of the tests and then discovered there was more wording incorrect plus tests that did not belong.

**Pull request closed without merging**
